packages/ckb-indexer/src/indexer.ts

```diff
@@ -0,0 +1,264 @@
+  push: boolean;
+}
+
+export declare type Terminator = (
+  index: number,
+  cell: Cell
+) => TerminatorResult;
+
+const DefaultTerminator: Terminator = () => {
+  return { stop: false, push: true };
+};
+
+export type HexNum = string;
+export type IOType = "input" | "output";
+export type Bytes32 = string;
+export type GetTransactionsResult = {
+  block_number: HexNum;
+  io_index: HexNum;
+  io_type: IOType;
+  tx_hash: Bytes32;
+  tx_index: HexNum;
+};
+export interface GetTransactionsResults {
+  last_cursor: string;
+  objects: GetTransactionsResult[];
+}
+
+export interface GetCellsResults {
+  lastCursor: string;
+  objects: Cell[];
+}
+
+export interface AdditionalOptions {
+  sizeLimit?: number;
+  order?: Order;
+  lastCursor?: string | undefined;
+}
+
+function defaultLogger(level: string, message: string) {
+  console.log(`[${level}] ${message}`);
+}
+
+export class CkbIndexer implements Indexer {
+  uri: string;
+
+  constructor(public ckbIndexerUrl: string, public ckbRpcUrl: string) {
+    this.uri = ckbRpcUrl;
+  }
+
+  private getCkbRpc(): RPC {
+    return new RPC(this.ckbRpcUrl);
+  }
+
+  async tip(): Promise<Tip> {
+    const res = await this.request("get_tip");
+    return res as Tip;
+  }
+
+  asyncSleep(timeout: number): Promise<void> {
+    return new Promise((resolve) => setTimeout(resolve, timeout));
+  }
+
+  async waitForSync(blockDifference = 0): Promise<void> {
+    const rpcTipNumber = parseInt(
+      (await this.getCkbRpc().get_tip_header()).number,
+      16
+    );
+    while (true) {
+      const indexerTipNumber = parseInt((await this.tip()).block_number, 16);
+      if (indexerTipNumber + blockDifference >= rpcTipNumber) {
+        return;
+      }
+      await this.asyncSleep(1000);
+    }
+  }
+
+  /*
+   * Additional note:
+   * Only accept lock and type parameters as `Script` type, along with `data` field in QueryOptions. Use it carefully!
+   * */
+  collector(queries: CkbQueryOptions): CellCollector {
+    return new IndexerCollector(this, queries);
+  }
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any, @typescript-eslint/explicit-module-boundary-types
+  private async request(
+    method: string,
+    params?: any,
+    ckbIndexerUrl: string = this.ckbIndexerUrl
+  ): Promise<any> {
+    const data = {
+      id: 0,
+      jsonrpc: "2.0",
+      method,
+      params,
+    };
+    const res: rpcResponse = await axios.post(ckbIndexerUrl, data);
+    if (res.status !== 200) {
+      throw new Error(`indexer request failed with HTTP code ${res.status}`);
+    }
+    if (res.data.error !== undefined) {
+      throw new Error(
+        `indexer request rpc failed with error: ${JSON.stringify(
+          res.data.error
+        )}`
+      );
+    }
+    return res.data.result;
+  }
+
+  public async getCells(
+    searchKey: SearchKey,
+    terminator: Terminator = DefaultTerminator,
+    {
+      sizeLimit = 0x100,
+      order = Order.asc,
+      lastCursor = undefined,
```

Comment:
```suggestion
```

Reply:
* changed sizeLimit to 100
* move params init to method body